lib/Dialect/Comb/CombFolds.cpp

```diff
@@ -788,6 +788,28 @@ static bool canonicalizeLogicalCstWithConcat(Operation *logicalOp,
   return true;
 }
 
+// Determines whether the inputs to a logical element are of opposite
+// comparisons and can lowered into a constant.
+static bool canCombineOppositeBinCmpIntoConstant(OperandRange operands) {
+  llvm::DenseSet<std::tuple<ICmpPredicate, Value, Value>> seenPredicates(
+      operands.size());
+
```

Comment:
nit: Probably allocating the number of operands is too much. Can we use SmallDenseMap?